src/main/java/touro/spf/Graph.java

```diff
@@ -1,4 +1,50 @@
 package touro.spf;
 
+import java.util.HashSet;
+
 public class Graph {
+    private HashSet<Node> nodes = new HashSet<>();
+
+    public void buildGraph(HashSet<Node> ns) {
```

Comment:
@rachel613g I just added this for testing, so feel free to use or ignore it

Reply:
Cool, thanks.